### DISCUSSION TITLE
Version of workspace in DSL

### DISCUSSION THREAD

**Discussion by synclpz** [structurizr/java#224] [Q&A] [answered]
As I see the model in JSON does support the `version` property which could be used to determine the source tag for deployed architecture in Structurizr repo. I did not find any support for this in DSL though. Is this intentionally made decision or just the lack of version support in DSL parser?

**Comment by simonbrowndotje** [comment, marked as answer]
That's what I've used the `version` property for too. It's not exposed via the DSL, but you can use a script to set it:

```
    !script groovy {
        workspace.version = "..."
    }
```

**Reply by synclpz** [reply]
Might it be you have a code snippet that you used to extract latest git tag name on current branch in groovy? :)

**Reply by synclpz** [reply]
I decided to make it via CI variables in GitLab CI:

```kotlin
var ver = "local"
val ciTag = System.getenv("CI_COMMIT_TAG")
val ciBranch = System.getenv("CI_COMMIT_BRANCH")
val ciMrBranch = System.getenv("CI_MERGE_REQUEST_SOURCE_BRANCH_NAME")
val ciCommitShortSha = System.getenv("CI_COMMIT_SHORT_SHA")

if (ciTag != null) {
    ver = ciTag
} else if (ciBranch != null) {
    ver = ciBranch + "-" + ciCommitShortSha
} else if (ciMrBranch != null) {
    ver = ciMrBranch + "-" + ciCommitShortSha
}

workspace.version = ver
workspace.name = workspace.name + " [v." + ver + "]"
```
